corehq/apps/app_manager/models.py

```diff
@@ -4125,7 +4125,7 @@ def validate_for_build(self):
                 'type': 'report config ref invalid',
                 'module': self.get_module_info()
             })
-        if not self.reports:
+        elif not self.reports:
```

Comment:
If the condition above is `True` then there is an error in reports config, then calling `self.reports` will raise an Exception. So making this `elif` meant it would be called only if there were no errors.